src/libraries/Common/src/System/Runtime/InteropServices/ComEventsMethod.cs

```diff
@@ -99,7 +100,7 @@ private void PreProcessSignature()
         /// Since multicast delegate's built-in chaining supports only chaining instances of the same type,
         /// we need to complement this design by using an explicit linked list data structure.
         /// </summary>
-        private readonly List<DelegateWrapper> _delegateWrappers = new List<DelegateWrapper>();
+        private List<DelegateWrapper> _delegateWrappers = new List<DelegateWrapper>();
```

Comment:
Does `List<T>` by you much at this point?  Now that you're creating a new instance each time, adding is allocate+copy+store and remove is allocate+copy+copy.  You might slim it down slightly by just using an array directly.

Reply:
That is true, but then we do need to handle the removal case and that can be far more cumbersome. The add is simple, but removal requires me to copy, null and then shift > delete index down by one. I can do that but I think it is going to require more code unless there is as simple - remove and shift for arrays?

Reply:
```C#
var newArr = new DelegateWrapper[oldArr.Length - 1];
oldArr.AsSpan(0, i).CopyTo(newArr);
oldArr.AsSpan(i + 1).CopyTo(newArr.AsSpan(i));
```
? Regardless, it's up to you.  I don't know how much the extra allocation for lists matter here, especially since we're creating a new array now on every mutation.

Reply:
ImmutableArray?

Reply:
I don’t believe that’s available at the later this code compiled into (it’s compiled into CoreLib amongst other places)

Reply:
> ```csharp
> var newArr = new DelegateWrapper[oldArr.Length - 1];
> oldArr.AsSpan(0, i).CopyTo(newArr);
> oldArr.AsSpan(i + 1).CopyTo(newArr.AsSpan(i));
> ```

Ah. I regularly forget going to `Span` for these sorts of things.

Reply:
(It could also just use Array.Copy instead of AsSpan(...).Copy.)